### PR TITLE
Band shading

### PR DESCRIPTION
**Pull request by maxmcd**
WIP

### CODE REVIEW

**Comment by maxmcd** [comment]
This is ready. I've edited the lines.html example with an hour-long band labelled "Weekend".

@evanemolo @jessipedia any thoughts on this api? Should the label allow a custom size/color/position? I took some inspirastion from the highcharts "bands": https://api.highcharts.com/highcharts/xAxis.plotBands

image

File: examples/lines.html [hunk 1]
```diff
@@ -33,12 +33,23 @@
 }
 
 // instantiate our graph!
+console.log(seriesData)
```
Comment (evanemolo) [inline examples/lines.html]: nit: remove logs

File: examples/lines.html [hunk 2]
```diff
@@ -33,12 +33,23 @@
...
+
+var d = new Date();
+d.setHours(d.getHours() + 1);
 
 var graph = new Rickshaw.Graph( {
 	element: document.getElementById("chart"),
 	width: 960,
 	height: 500,
 	renderer: 'line',
+	bands: [{
```
Comment (evanemolo) [inline examples/lines.html]: nit: tabs

File: src/js/Rickshaw.Graph.js [hunk 3]
```diff
@@ -125,6 +126,16 @@ Rickshaw.Graph = function(args) {
 		this.series.active = function() { return self.series.filter( function(s) { return !s.disabled } ) };
 	};
 
+	this.setBands = function(bands) {
+		// TODO: validate bands
```
Comment (evanemolo) [inline src/js/Rickshaw.Graph.js]: nit: can you add some context to this comment?

File: src/js/Rickshaw.Graph.Band.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+Rickshaw.namespace("Rickshaw.Graph.Band");
```
Comment (evanemolo) [inline src/js/Rickshaw.Graph.Band.js]: should this be namespaced under Rickshaw.Graph.Render?

File: src/js/Rickshaw.Graph.js [hunk 5]
```diff
@@ -69,6 +69,7 @@ Rickshaw.Graph = function(args) {
 	};
 
 	this._loadRenderers = function() {
+		this._bandRenderer = new Rickshaw.Graph.Band( { graph: self } );
```
Comment (evanemolo) [inline src/js/Rickshaw.Graph.js]: per the related question in Band.js,  it looks like line 78 would do this work for you if it's namespace to Renderer. With that said, I have not pulled the code down, so I'm not sure if that's possible.

File: src/js/Rickshaw.Graph.Band.js (hunk 4 above)
Comment (maxmcd) [reply src/js/Rickshaw.Graph.Band.js]: As far as I can tell the renders are all line renderers, they're loaded into an object and can be used based on how you want to render plotted lines. 

I originally thought about putting this under Render, but Band rendering doesn't need any of the underlying utility functions that are provided by the parent class. 

Renderers also assume they're the only thing rendering during a re-paint and completely clear the graph. So just adding two renderers to the same structure naively would only paint the most recently rendered data (band or lines).

**Comment by evanemolo** [comment]
Nice work - from the screen shot this looks great.

File: examples/lines.html (hunk 2 above)
Comment (evanemolo) [reply examples/lines.html]: We can update the jshint config in another PR. I would just leave as-is for now.

**Pull request merged**